pkg/scale/decode.go

```diff
@@ -426,6 +428,34 @@ func (ds *decodeState) decodeArray(dstv reflect.Value) (err error) {
 	return
 }
 
+func (ds *decodeState) decodeMap(dstv reflect.Value) (err error) {
+	l, err := ds.decodeLength()
+	if err != nil {
+		return
```

Comment:
nit add error context so we can trace back what failed and which code path

```suggestion
		return fmt.Errorf("decoding length: %w", err)
```